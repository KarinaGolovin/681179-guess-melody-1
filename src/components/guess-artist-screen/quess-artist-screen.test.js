import React from 'react';
import renderer from 'react-test-renderer';
import GuessArtistScreen from './guess-artist-screen.jsx';

const mock = {
  song: {
    artist: `Test`,
    trackSrc: ``,
  },
  answers: [
    {
      artist: `Test`,
      picture: ``
    },
    {
      artist: `One`,
      picture: ``
    },
    {
      artist: `Two`,
      picture: ``
    },
    {
      artist: `Three`,
      picture: ``
    },
  ],
};

it(`GuessArtistScreen correctly rendered`, () => {
  const component = renderer.create(
      <GuessArtistScreen
        question={mock}
        onAnswer={jest.fn()}
      />,
      {
        createNodeMock: () => ({
        })
      }
  ).toJSON();

  expect(component).toMatchSnapshot();
});
