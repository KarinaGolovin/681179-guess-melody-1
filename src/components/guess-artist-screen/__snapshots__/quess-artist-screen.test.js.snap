// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessArtistScreen correctly rendered 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      type="button"
    />
    <audio
      src=""
    />
  </div>
  <form
    className="game__artist"
    onChange={[Function]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Test"
          className="artist__picture"
          src=""
        />
        Test
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="One"
          className="artist__picture"
          src=""
        />
        One
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Two"
          className="artist__picture"
          src=""
        />
        Two
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-3"
        name="answer"
        type="radio"
        value="answer-3"
      />
      <label
        className="artist__name"
        htmlFor="answer-3"
      >
        <img
          alt="Three"
          className="artist__picture"
          src=""
        />
        Three
      </label>
    </div>
  </form>
</section>
`;
